src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts

```diff
@@ -143,13 +145,18 @@ export class InstructorCourseDetailsPageComponent implements OnInit {
         this.loadPrivilege(courseid, key, data);
       });
 
-      if (!sections.length) {
-        this.isStudentsLoading = false;
+      if (!Object.keys(sections).length) {
```

Comment:
I just tried this PR out. Looks like this single line fix alone can solve the issue. Are the other changes neccessary?

Reply:
Hmmm, yup seems like it. Initially, I thought that the check in the `*ngIf` was the issue. When I change from using `students.length === 0` to a boolean variable, it works also.

I will update it.

Reply:
@lyeyixian Have you removed the unnecessary changes?

Reply:
Yes, I already did. You can see it in this commit https://github.com/TEAMMATES/teammates/pull/11271/commits/00d5f0dac86b6649bde712371abcab1d92fc5248